src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs

```diff
@@ -2,43 +2,42 @@
 // Licensed under the MIT license.
 
 using System;
-using System.Diagnostics.Contracts;
+using System.Collections.Generic;
+using System.Linq;
 using Microsoft.PowerFx.Core.Types;
 using Microsoft.PowerFx.Core.Utils;
 
 namespace Microsoft.PowerFx.Types
 {
-    public class RecordType : AggregateType
+    /// <summary>
+    /// Represents a Record type within PowerFx. If this is subclassed, it's quite likely that 
+    /// <see cref="RecordValue"/> should be as well. 
+    /// If the type is known in advance and easy to construct, use <see cref="KnownRecordType"/> instead of
+    /// deriving from this. 
+    /// </summary>
+    public abstract class RecordType : AggregateType
     {
+        // The internal constructor allows us to wrap known DTypes, while the public constructor
+        // will create a DType that wraps derived TryGetField/Fields/Identity calls
         internal RecordType(DType type)
             : base(type)
         {
-            Contract.Assert(type.IsRecord);
+            Contracts.Assert(type.IsRecord);
         }
 
-        public RecordType()
-            : base(DType.EmptyRecord)
```

Comment:
This is a significant breaking change (removing a default ctor). 
In hindsite, we should have used a static builder.

Reply:
wow, I can see a bunch of test updates here... that's definitely a signal about the amount of partner breaks we may expect.